tests/semantic_analyser.cpp

```diff
@@ -4795,21 +4812,30 @@ BEGIN { unroll(1) { $a = 1; } print(($a)); }
 TEST(semantic_analyser, invalid_assignment)
 {
   test_error("BEGIN { @a = hist(10); let $b = @a; }", R"(
-stdin:1:24-35: ERROR: Map value 'hist_t' cannot be assigned to a scratch variable.
+stdin:1:24-35: ERROR: Value 'hist_t' cannot be assigned to a scratch variable.
 BEGIN { @a = hist(10); let $b = @a; }
                        ~~~~~~~~~~~
+stdin:1:24-30: WARNING: Variable $b never assigned to.
+BEGIN { @a = hist(10); let $b = @a; }
+                       ~~~~~~
 )");
 
   test_error("BEGIN { @a = lhist(123, 0, 123, 1); let $b = @a; }", R"(
-stdin:1:37-48: ERROR: Map value 'lhist_t' cannot be assigned to a scratch variable.
+stdin:1:37-48: ERROR: Value 'lhist_t' cannot be assigned to a scratch variable.
 BEGIN { @a = lhist(123, 0, 123, 1); let $b = @a; }
                                     ~~~~~~~~~~~
+stdin:1:37-43: WARNING: Variable $b never assigned to.
+BEGIN { @a = lhist(123, 0, 123, 1); let $b = @a; }
+                                    ~~~~~~
 )");
 
   test_error("BEGIN { @a = stats(10); let $b = @a; }", R"(
-stdin:1:25-36: ERROR: Map value 'stats_t' cannot be assigned to a scratch variable.
+stdin:1:25-36: ERROR: Value 'stats_t' cannot be assigned to a scratch variable.
 BEGIN { @a = stats(10); let $b = @a; }
                         ~~~~~~~~~~~
+stdin:1:25-31: WARNING: Variable $b never assigned to.
```

Comment:
This is kinda funny. Side effect of the map sugar pass?

Reply:
I think it's just because the type is still none at the end. I can chase this down, but as long as the error messages are sensible I'm not generally fighting the errors.

Reply:
It's not big enough to block, it's just weird.